src/libAtoms/linearalgebra.f95

```diff
@@ -3723,6 +3723,38 @@ subroutine matrix_z_print_mathematica(label, this, verbosity, file)
     call print ("};", verbosity, file)
   end subroutine matrix_z_print_mathematica
 
+  subroutine matrix_condition_number(A, rcond, norm)
+    real(dp), intent(inout) :: A(:,:)
+    real(dp), intent(out) :: rcond
+    character, intent(in) :: norm !% 1 or O: 1-norm, I: inf-norm
```

Comment:
I don't understand why norm is needed - isn't condition numbers just `maximum(eigvals)/minimum(eigvals)`?

Reply:
That is a rabbit hole of its own. You can have a condition number for different properties (usually inversion) and for different norms. Each norm seems to tell a slightly different story.

Reply:
LAPACK/SCALAPACK only support condition numbers for 1-norm and inf-norm. Norms in general also include the 2-norm and the Frobenius norm I think.

Reply:
I suggest this routine should stay general but we only use one of the options. The choice of norm for gap_fit could be set via the option, so instead of `do_condition_number=T` it could be `condition_number_norm=I`.

Reply:
agreed - but `condition_number_norm=inf` better than `I` since `I` looks like an identity matrix.

Reply:
although I see the appeal of a single character input - could switch to `character(3)` and use `abs` and `inf` for 1 and inf-norm respectively

Reply:
Or follow `np.linalg.cond()` semantics, since they will be familiar to most people. Not a big deal either way.

Reply:
Can do. Right now it just pipes it to (Sca)LAPACK. Conversion just for the gap_fit option or for the condition routine itself?

On the other hand, the identity matrix is not a norm so confusion is less likely.

Reply:
OK, fine, stick with what you have. It's a Fortran code after all so let's follow (Sca)LAPACK.

Reply:
But norm choice via clo?

Reply:
Yes

Reply:
Both `do_condition_number=T` and `condition_number_norm=1` or just the latter with special input for off (e.g. space ' ' or 'X')?

Reply:
Why does ParamReader enforce character values to be of length STRING_LENGTH?

Reply:
I have checked that register_params accepts empty string options. It does so I chose to only keep `condition_number_norm`.

Reply:
looks good, thanks